build.gradle.kts

```diff
@@ -20,9 +22,41 @@ configure<ApiValidationExtension> {
     nonPublicMarkers.add("pbandk.PbandkInternal")
 }
 
+val sonatypeApiUser = providers.gradlePropertyOrEnvironmentVariable("sonatypeApiUser")
+val sonatypeApiKey = providers.gradlePropertyOrEnvironmentVariable("sonatypeApiKey")
+if (sonatypeApiUser.isPresent && sonatypeApiKey.isPresent) {
+    configure<NexusPublishExtension> {
```

Comment:
It turns out that publishing a release to Maven Central is a bit more involved than just uploading the artifacts to the OSSRH maven repository via gradle's `maven-publish` plugin. Every upload to OSSRH creates a "staging repository". You then need to "close" and "release" this repository before it'll get pushed to Maven Central. I found this `nexus-publish` gradle plugin which will take care of this process for you.

Reply:
Yeah, we've considered this as well. 

I think this is also very much a process question: would you like to test artifacts before having them publicly on Maven Central essentially forever? I can see the arguments both ways honestly...

An approach could be to stick with the manual steps for now, see how annoying they are (or how frequent release are) and optimize from there onwards.

Reply:
True there are advantages to both approaches. I didn't want to force everyone else on my team to have to know all the details about how to login to Sonatype and do the manual release process. Switching to the `nexus-publish` plugin was the easier route 😄 I'm also a big believer that `master` should always be in a releasable state. In which case, we shouldn't need to do any manual testing prior to making a release. Just creating a tag in `git` should be sufficient.